analytics.ts

```diff
@@ -0,0 +1,121 @@
+import * as Settings from '@taqueria/protocol/Settings';
+import { chain, map, resolve } from 'fluture';
+import { pipe } from 'https://deno.land/x/fun@v1.0.0/fns.ts';
+import { getMachineId } from 'https://deno.land/x/machine_id@v0.3.0/mod.ts';
+import type { DenoArgs } from './taqueria-types.ts';
+import * as utils from './taqueria-utils/taqueria-utils.ts';
+
+const {
+	doesPathExist,
+	readJsonFile,
+	writeJsonFile,
+	eager,
+} = utils.inject({
+	stdout: Deno.stdout,
+	stderr: Deno.stderr,
+});
+
+const consentPrompt =
+	'Allow Taqueria anonymously report usage statistics, in accordance with the privacy policy to help make Taqueria better? [y/yes] or [n/no]';
+const optInConfirmationPrompt =
+	'Allow Taqueria anonymously report usage statistics, in accordance with the privacy policy to help make Taqueria better? [y/yes] or [n/no]';
+const optOutConfirmationPrompt = 'Are you sure you want to turn off usage statistic reporting? [y/yes] or [n/no]';
+export const settingsFilePath = Deno.env.get('HOME') + '/.taq-settings.json';
```

Comment:
Jimmy, Deno.env should be used via dependency injection. That's why in cli.ts you'll see env sent as an argument:
```ts
export const run = (env: EnvVars, inputArgs: DenoArgs, i18n: i18n.t) => {
...
}
```

We'll need that for testing. You can look at the pattern we use with plugins.ts or taqueria-utils.ts to replicate.